python/pyspark/pandas/indexing.py

```diff
@@ -656,16 +658,20 @@ def __setitem__(self, key, value):
                 and (isinstance(self, iLocIndexer) or not same_anchor(value, self._psdf_or_psser))
             ):
                 psdf = self._psdf_or_psser.copy()
```

Comment:
We can use `cast` for `psdf`, then we can remove `ignore` in the following lines.